crates/common/tedge_utils/src/inotify.rs

```diff
@@ -0,0 +1,17 @@
+pub use inotify_crate::EventMask;
+pub use inotify_crate::WatchMask;
+use inotify_crate::{EventStream, Inotify};
+use std::path::Path;
+
+pub fn inofity_file_watch_stream(
+    config_file: &Path,
+    watch_flags: WatchMask,
+) -> Result<EventStream<[u8; 1024]>, anyhow::Error> {
+    let buffer = [0; 1024];
+    let mut inotify = Inotify::init().expect("Error while initializing inotify instance");
+
+    inotify
+        .add_watch(config_file, watch_flags)
+        .expect("Failed to add file watch");
```

Comment:
This `expect` must be replaced by a proper error handler. The file might be non-accessible or the path might have a typo.